### PR TITLE
Week 5

### PR DESCRIPTION
**Pull request by VanillaMint02**
Week-5 assignment

### CODE REVIEW

**Comment by VanillaMint02** [comment]
issue : I could not make it work without those services in the auth module.

File: apps/api/src/app/app.module.ts [hunk 1]
```diff
@@ -24,8 +25,9 @@ import { UserModule } from './user/user.module';
     }),
     QuestionManagementModule,
     UserModule,
+    AuthModule,
   ],
   controllers: [],
-  providers: [],
+  providers: [AuthModule],
```
Comment (antonio-marasescu) [inline apps/api/src/app/app.module.ts]: **issue:** an AuthModule is not a provider, remove it from here

File: apps/api/src/app/auth/strategies/local.strategy.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import { UserDto } from '../../user/dtos/user.dto';
+
+@Injectable()
+export class LocalStrategy extends PassportStrategy(Strategy) {
+  constructor(private authService: AuthService) {
+    super({ usernameField: 'email', passwordField: 'password' });
+  }
+
+  async validate(email: string, password: string): Promise<UserDto> {
+    console.log('email is : ' + email + ' password is : ' + password);
```
Comment (antonio-marasescu) [inline apps/api/src/app/auth/strategies/local.strategy.ts]: **issue:** do not keep `console.log` anywhere in your code, at best use the `Logger` class here, but in your specific case you are also exposing the user password here. My suggestion is to remove this line

File: apps/api/src/app/question-management/dtos/answer.dto.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export class AnswerDto {
+  id?: string;
+  content: string;
```
Comment (antonio-marasescu) [inline apps/api/src/app/question-management/dtos/answer.dto.ts:15]: **suggestion:** As reminder, please annotate all your DTO's with the Swagger annotation (description and example)

**Review by antonio-marasescu — APPROVED** [review]

**Pull request merged**